plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -517,11 +721,7 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
    bool                                              _disable_subjective_p2p_billing              = true;
    bool                                              _disable_subjective_api_billing              = true;
    fc::time_point                                    _irreversible_block_time;
-   fc::time_point                                    _accepted_block_time;
    bool                                              _is_savanna_active                           = false;
-   bool                                              _is_producer_active_finalizer                = false;
-   // are there any active finalizers that are not part of producers of this node (not expected to be false outside tests)
-   bool                                              _other_active_finalizers                     = false;
```

Comment:
The tracking of active finalizers is rather expensive, seems like we should avoid tracking them on validation nodes. I've wrapped your `production_pause_vote_trakcer` in `implicit_production_pause_vote_tracker`. Also add a `pause_check` enum to `check_pause_status` since which `vote_timing` to check depends on active finalizers.